crates/concurrency/concurrency.rs

```diff
@@ -0,0 +1,51 @@
+use crossbeam::channel::{Sender, select_biased, unbounded};
```

Comment:
Let's add a comment explaining why we're doing our own threadpool. Explain the problems we found and that we preferred a simpler solution that helped us model correctly how things are going to work.
Otherwise future us will be tempted to delete this and use rayon, going full circle.